inductiva/molecules/scenarios/protein_solvation/protein_solvation.py

```diff
@@ -97,80 +70,87 @@ def gen_config(self, simulator: Simulator):
+                       o="protein.gro",
+                       water="tip3p",
+                       user_input="6"))
+    pipeline.append(
+        GROMACSCommand(method_name="editconf",
+                       f="protein.gro",
+                       o="protein_box.gro",
+                       c="yes",
+                       d="1.0",
+                       bt="cubic"))
+    pipeline.append(
+        GROMACSCommand(method_name="genbox",
+                       cp="protein_box.gro",
+                       o="protein_solv.gro",
+                       p="topol.top"))
+    pipeline.append(
+        GROMACSCommand(method_name="grompp",
+                       f="ions.mdp",
+                       c="protein_solv.gro",
+                       p="topol.top",
+                       o="ions.tpr"))
+    pipeline.append(
+        GROMACSCommand(method_name="genion",
+                       s="ions.tpr",
+                       o="protein_solv_ions.gro",
+                       p="topol.top",
+                       pname="NA",
+                       nname="CL",
+                       neutral="yes"))
     # Energy minimization
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="grompp",
-                  f="energy_minimization.mdp",
-                  c="protein_solv_ions.gro",
-                  p="topol.top",
-                  o="em.tpr")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="mdrun",
-                  deffnm="em",
-                  v="yes")
+    pipeline.append(
+        GROMACSCommand(method_name="grompp",
+                       f="energy_minimization.mdp",
+                       c="protein_solv_ions.gro",
+                       p="topol.top",
+                       o="em.tpr"))
+    pipeline.append(GROMACSCommand(method_name="mdrun", deffnm="em", v="yes"))
+    pipeline.append(
+        GROMACSCommand(method_name="grompp",
+                       f="simulation.mdp",
+                       c="em.gro",
+                       r="em.gro",
+                       p="topol.top",
+                       o="solvated_protein.tpr"))
     # Simulation
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="grompp",
-                  f="simulation.mdp",
-                  c="em.gro",
-                  r="em.gro",
-                  p="topol.top",
-                  o="solvated_protein.tpr")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="mdrun",
-                  deffnm="solvated_protein",
-                  v="yes")
+    pipeline.append(
+        GROMACSCommand(method_name="mdrun", deffnm="solvated_protein", v="yes"))
+    return pipeline
 
 
 @ProteinSolvation.gen_config.register
```

Comment:
We should consider renaming this to `gen_config_files`, since it can in general be used to generate an arbitrary number of config files. Should we open an issue to address it in a future PR?

Reply:
Yes!

Reply:
Agreed :)